engine/runtime-with-polyglot/src/test/scala/org/enso/interpreter/test/instrument/RuntimeVisualizationsTest.scala

```diff
@@ -3415,4 +3415,266 @@ class RuntimeVisualizationsTest
     }
     new String(data1, StandardCharsets.UTF_8) shouldEqual "C"
   }
+
+  it should "execute expression in the scope of local expression cached" in {
+    val contextId       = UUID.randomUUID()
+    val requestId       = UUID.randomUUID()
+    val visualizationId = UUID.randomUUID()
+    val moduleName      = "Enso_Test.Test.Main"
+    val metadata        = new Metadata
+
+    val idOp1 = metadata.addItem(23, 2)
+    val idOp2 = metadata.addItem(42, 13)
+
+    val code =
+      """main =
+        |    operator1 = 42
+        |    operator2 = operator1 + 1
+        |    operator2
+        |
+        |fun1 x = x.to_text
+        |""".stripMargin.linesIterator.mkString("\n")
+    val contents = metadata.appendToCode(code)
```

Comment:
Can we use `metadata.assertInCode` here so I know what part of the code `idOp1` and `idOp2` represent?